pkg/filesystem/subvolume.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subvolume
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	exec "github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type fsStruct struct {
+	Name string
+	data []string
+}
+
+type subVolumeInfo struct {
+	svg string
+	fs  string
+}
+
+func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace, includeStale string) map[string]subVolumeInfo {
+
+	pvList, err := clientsets.Kube.CoreV1().PersistentVolumes().List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Error fetching PVs: %v\n", err))
+		os.Exit(1)
+	}
+	subvolumeNames := make(map[string]subVolumeInfo)
+	for _, pv := range pvList.Items {
+		if pv.Spec.CSI.VolumeAttributes["subvolumeName"] != "" {
+			subvolumeNames[pv.Spec.CSI.VolumeAttributes["subvolumeName"]] = subVolumeInfo{}
+		}
+	}
+
+	return listCephFSSubvolumes(ctx, clientsets, operatorNamespace, clusterNamespace, includeStale, subvolumeNames)
+
+}
+
+func listCephFSSubvolumes(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace, includeStale string, subvolumeNames map[string]subVolumeInfo) map[string]subVolumeInfo {
```

Comment:
Always use a bool if possible, it's easier than comparing strings that may have case-sensitive issues
```suggestion
func listCephFSSubvolumes(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace, includeStale bool, subvolumeNames map[string]subVolumeInfo) map[string]subVolumeInfo {
```

Reply:
We are getting the flag as string and I don't see extra conversion to bool, even this same pattern is used elsewhere. If you still thing we should add it as bool, let me know will update it.